sysl2/sysl/seqs/visitor.go

```diff
@@ -333,21 +379,24 @@ func (v *SequenceDiagramVisitor) visitEndpoint(e *EndpointElement) error {
 
 	if len(endpoint.Stmt) > 0 {
 		visiting := fmt.Sprintf("%s <- %s", e.appName, e.endpointName)
-		comment, hitUpto := e.uptos[visiting]
+		upto, hitUpto := e.uptos[visiting]
+		if hitUpto && e.uptos[visiting].valueType >= BB_APPLICATION {
```

Comment:
```go
		upto, has := e.uptos[visiting]; has && upto.valueType >= BB_APPLICATION {
```

Reply:
The variable hitUpto has a higher scope and is shared with other if statements in the outer if len(endpoint.Stmt) > 0 { block.

Reply:
The variables upto and hitUpto are shared with other if statements.